Sources/Sentry/SentryBaseIntegration.m

```diff
@@ -144,8 +144,8 @@ - (BOOL)shouldBeEnabledWithOptions:(SentryOptions *)options
 
     if (integrationOptions & kIntegrationOptionEnableReplay) {
         if (@available(iOS 16.0, tvOS 16.0, *)) {
-            if (options.sessionReplayOptions.replaysOnErrorSampleRate == 0
-                && options.sessionReplayOptions.replaysSessionSampleRate == 0) {
+            if (options.experimental.sessionReplayOptions.errorSampleRate == 0
```

Comment:
I'm a bit confused because originally it was `replaysOnErrorSampleRate` and it's renaming away from it.

@romtsn lets stay all with the same name as web? `replaysOnErrorSampleRate` this can be set directly on `SentryOptions` then also to simplify

Reply:
On a chat with Dhiogo I got convinced that it's better to have the replay object.
So we make sure it can't be null, and it's easier to navigate:

```
options.sessionReplay.errorSampleRate=1.0
options.sessionReplay.sessionSampleRate=1.0
```